pkg/vxlan/vxlan.go

```diff
@@ -0,0 +1,271 @@
+	if errors.Is(err, syscall.EEXIST) {
+		return nil
+	}
+
+	return errors.Wrapf(err, "unable to configure address %q on vxlan device %q", ipAddress, i.link.Name)
+}
+
+func (i *Interface) SetupLink() error {
+	return i.netLink.LinkSetUp(i.link) //nolint:wrapcheck // No need to wrap here.
+}
+
+func (i *Interface) DeleteLinkDevice() error {
+	err := i.netLink.LinkDel(i.link)
+	return errors.Wrapf(err, "error deleting vxlan device %q", i.link.Name)
+}
+
+func (i *Interface) AddFDB(ipAddress net.IP, hwAddr string) error {
+	macAddr, err := net.ParseMAC(hwAddr)
+	if err != nil {
+		return errors.Wrapf(err, "invalid MAC Address %q", hwAddr)
+	}
+
+	neigh := &netlink.Neigh{
+		LinkIndex:    i.link.Index,
+		Family:       unix.AF_BRIDGE,
+		Flags:        netlink.NTF_SELF,
+		Type:         netlink.NDA_DST,
+		IP:           ipAddress,
+		State:        netlink.NUD_PERMANENT | netlink.NUD_NOARP,
+		HardwareAddr: macAddr,
+	}
+
+	err = i.netLink.NeighAppend(neigh)
+	if err != nil {
+		return errors.Wrapf(err, "unable to add the bridge fdb entry %v", neigh)
+	}
+
+	logger.V(log.DEBUG).Infof("Successfully added the bridge fdb entry %v", neigh)
+
+	return nil
+}
+
+func (i *Interface) DelFDB(ipAddress net.IP, hwAddr string) error {
+	macAddr, err := net.ParseMAC(hwAddr)
+	if err != nil {
+		return errors.Wrapf(err, "invalid MAC Address %q", hwAddr)
+	}
+
+	neigh := &netlink.Neigh{
+		LinkIndex:    i.link.Index,
+		Family:       unix.AF_BRIDGE,
+		Flags:        netlink.NTF_SELF,
+		Type:         netlink.NDA_DST,
+		IP:           ipAddress,
+		State:        netlink.NUD_PERMANENT | netlink.NUD_NOARP,
+		HardwareAddr: macAddr,
+	}
+
+	err = i.netLink.NeighDel(neigh)
+	if err != nil {
+		return errors.Wrapf(err, "unable to delete the bridge fdb entry %v", neigh)
+	}
+
+	logger.V(log.DEBUG).Infof("Successfully deleted the bridge fdb entry %v", neigh)
+
+	return nil
+}
+
+func (i *Interface) AddRoutes(gwIP, srcIP net.IP, tableID int, destCIDRs ...net.IPNet) error {
+	for j := range destCIDRs {
+		route := &netlink.Route{
+			LinkIndex: i.link.Index,
+			Src:       srcIP,
+			Dst:       &destCIDRs[j],
+			Gw:        gwIP,
+			Type:      netlink.NDA_DST,
+			Flags:     netlink.NTF_SELF,
+			Priority:  100,
+			Table:     tableID,
+		}
+		err := i.netLink.RouteAdd(route)
+
+		if errors.Is(err, syscall.EEXIST) {
+			err = i.netLink.RouteReplace(route)
+		}
+
+		if err != nil {
+			return errors.Wrapf(err, "unable to add the route entry %v", route)
+		}
+
+		logger.V(log.DEBUG).Infof("Successfully added the route entry %v and gw ip %v", route, gwIP)
+	}
+
+	return nil
+}
+
+func (i *Interface) DelRoutes(tableID int, destCIDRs ...net.IPNet) error {
+	for j := range destCIDRs {
+		route := &netlink.Route{
+			LinkIndex: i.link.Index,
+			Dst:       &destCIDRs[j],
+			Gw:        nil,
+			Type:      netlink.NDA_DST,
+			Flags:     netlink.NTF_SELF,
+			Priority:  100,
+			Table:     tableID,
+		}
+
+		err := i.netLink.RouteDel(route)
```

Comment:
Don't we want to return nil if route doesn't exist ? ,  like 
https://github.com/submariner-io/submariner/blob/devel/pkg/routeagent_driver/handlers/ovn/gateway_dataplane.go#L48

Reply:
yeah that should be fine here.